### ISSUE TITLE
Bleadperl v5.15.4-249-ga03f0b9 breaks RKITOVER/String-ToIdentifier-EN-0.06.tar.gz

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11767]
Migrated from [rt.perl.org#104226](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=104226) (status was 'resolved')

Searchable as RT104226$

**Comment by p5pRT** [comment]
### From @trwyant

#### Created by @trwyant



Under the default build of Perl 5\.15\.5&#8203;:

$ perl \-Mcharnames \-e ''
Can't locate unicore/Name\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: /usr/local/perl/5\.15\.5/lib/site\_perl/5\.15\.5/darwin\-2level /usr/local/perl/5\.15\.5/lib/site\_perl/5\.15\.5 /usr/local/perl/5\.15\.5/lib/5\.15\.5/darwin\-2level /usr/local/perl/5\.15\.5/lib/5\.15\.5 \.\) at /usr/local/perl/5\.15\.5/lib/5\.15\.5/charnames\.pm line 6\.
BEGIN failed\-\-compilation aborted at /usr/local/perl/5\.15\.5/lib/5\.15\.5/charnames\.pm line 6\.
Compilation failed in require\.
BEGIN failed\-\-compilation aborted\.

What appears to have happened is that lib/unicore/Name\.pm was generated\,
but the 'make install' never installed it\.  There were no test failures
when building this Perl\. Specifically\, lib/charnames passed\. I can
supply build and test logs if desired\. FYI\,

$ ls /usr/local/perl/5\.15\.5/lib/5\.15\.5/unicore/
Blocks\.txt         Decomposition\.pl   NamedSequences\.txt UCD\.pl
CaseFolding\.txt    Heavy\.pl           SpecialCasing\.txt  lib
CombiningClass\.pl  Name\.pl            To                 version

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=high
    module=charnames

Site configuration information for perl 5.15.5:

Configured by tom at Mon Nov 21 09:19:16 EST 2011.

Summary of my perl5 (revision 5 version 15 subversion 5) configuration:
   
  Platform:
    osname=darwin, osvers=9.8.0, archname=darwin-2level
    uname='darwin wyants-white-too.local 9.8.0 darwin kernel version 9.8.0: wed jul 15 16:55:01 pdt 2009; root:xnu-1228.15.4~1release_i386 i386 '
    config_args='-Dprefix=/usr/local/perl/5.15.5'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include',
    optimize='-O3',
    cppflags='-fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.0.1 (Apple Inc. build 5484)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='env MACOSX_DEPLOYMENT_TARGET=10.3 cc', ldflags ='-fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /opt/local/lib /usr/lib
    libs=-lgdbm -ldbm -ldl -lm -lutil -lc
    perllibs=-ldl -lm -lutil -lc
    libc=, so=dylib, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=bundle, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags=' -bundle -undefined dynamic_lookup -L/usr/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.15.5:
    /usr/local/perl/5.15.5/lib/site_perl/5.15.5/darwin-2level
    /usr/local/perl/5.15.5/lib/site_perl/5.15.5
    /usr/local/perl/5.15.5/lib/5.15.5/darwin-2level
    /usr/local/perl/5.15.5/lib/5.15.5
    .


Environment for perl 5.15.5:
    DYLD_LIBRARY_PATH (unset)
    HOME=/Users/tom
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/perl/5.15.5/bin:/opt/local/bin:/opt/local/sbin:/ImageMagick-6.4.1/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/X11/bin:/Users/tom/Code/Tools
    PERLPKGOPT=-noppm
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @andk

Also affected in the same or in a similar way are
JKEGL/Marpa\-PP\-0\.011\_002\.tar\.gz\, RSCHUPP/PAR\-Packer\-1\.010\.tar\.gz\,
ZAG/Perl6\-Pod\-0\.35\.tar\.gz\, DCONWAY/Regexp\-Grammars\-1\.014\.tar\.gz and more\.

git bisect


---
&nbsp;  a03f0b9fc88f383466124c0fba3dd41d5c76b99e is the first bad commit
&nbsp;  commit a03f0b9fc88f383466124c0fba3dd41d5c76b99e
&nbsp;  Author&#8203;: Karl Williamson \<public@&#8203;khwilliamson\.com>
&nbsp;  Date&#8203;:   Tue Oct 4 11&#8203;:58&#8203;:52 2011 \-0600

&nbsp;      Use newly created Name\.pm file

&nbsp;      Name\.pm has been populated in an earlier commit\.  This removes the
&nbsp;      now redundant information in Name\.pl\, and changes charnames to include
&nbsp;      the new Name\.pm

diagnostics


---
&nbsp;  t/00\_basic\.t \.\.\.\.\.\.\.\.\.\.\. Can't locate unicore/Name\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: t/
&nbsp;  lib /tmp/tmp\.JTEOUvrRxA/String\-ToIdentifier\-EN\-0\.06\-OWprHR/inc /tmp/tmp\.JTEOUvrR
&nbsp;  xA/String\-ToIdentifier\-EN\-0\.06\-OWprHR/blib/lib /tmp/tmp\.JTEOUvrRxA/String\-ToIden
&nbsp;  tifier\-EN\-0\.06\-OWprHR/blib/arch /home/src/perl/repoperls/installed\-perls/perl/v5
&nbsp;  \.15\.4\-249\-ga03f0b9/a2da/lib/site\_perl/5\.15\.4/x86\_64\-linux\-thread\-multi\-ld /home/
&nbsp;  src/perl/repoperls/installed\-perls/perl/v5\.15\.4\-249\-ga03f0b9/a2da/lib/site\_perl/
&nbsp;  5\.15\.4 /home/src/perl/repoperls/installed\-perls/perl/v5\.15\.4\-249\-ga03f0b9/a2da/l
&nbsp;  ib/5\.15\.4/x86\_64\-linux\-thread\-multi\-ld /home/src/perl/repoperls/installed\-perls/
&nbsp;  perl/v5\.15\.4\-249\-ga03f0b9/a2da/lib/5\.15\.4 \.\) at /home/src/perl/repoperls/install
&nbsp;  ed\-perls/perl/v5\.15\.4\-249\-ga03f0b9/a2da/lib/5\.15\.4/charnames\.pm line 6\.
&nbsp;  BEGIN failed\-\-compilation aborted at /home/src/perl/repoperls/installed\-perls/pe
&nbsp;  rl/v5\.15\.4\-249\-ga03f0b9/a2da/lib/5\.15\.4/charnames\.pm line 6\.
&nbsp;  Compilation failed in require at /home/src/perl/repoperls/installed\-perls/perl/v
&nbsp;  5\.15\.4\-249\-ga03f0b9/a2da/lib/5\.15\.4/Unicode/UCD\.pm line 6\.
&nbsp;  BEGIN failed\-\-compilation aborted at /home/src/perl/repoperls/installed\-perls/pe
&nbsp;  rl/v5\.15\.4\-249\-ga03f0b9/a2da/lib/5\.15\.4/Unicode/UCD\.pm line 6\.
&nbsp;  Compilation failed in require at /tmp/tmp\.JTEOUvrRxA/String\-ToIdentifier\-EN\-0\.06
&nbsp;  \-OWprHR/blib/lib/String/ToIdentifier/EN\.pm line 8\.      
&nbsp;  BEGIN failed\-\-compilation aborted at /tmp/tmp\.JTEOUvrRxA/String\-ToIdentifier\-EN\-
&nbsp;  0\.06\-OWprHR/blib/lib/String/ToIdentifier/EN\.pm line 8\.  
&nbsp;  Compilation failed in require at t/lib/TestString\.pm line 7\.
&nbsp;  BEGIN failed\-\-compilation aborted at t/lib/TestString\.pm line 7\.
&nbsp;  Compilation failed in require at t/00\_basic\.t line 5\.   
&nbsp;  BEGIN failed\-\-compilation aborted at t/00\_basic\.t line 5\.
&nbsp;  t/00\_basic\.t \.\.\.\.\.\.\.\.\.\.\. Dubious\, test returned 2 \(wstat 512\, 0x200\)
&nbsp;  No subtests run                                         

perl \-V


---
&nbsp;  Summary of my perl5 \(revision 5 version 15 subversion 4\) configuration&#8203;:
&nbsp;    Commit id&#8203;: a03f0b9fc88f383466124c0fba3dd41d5c76b99e
&nbsp;    Platform&#8203;:
&nbsp;      osname=linux\, osvers=2\.6\.32\-5\-amd64\, archname=x86\_64\-linux\-thread\-multi\-ld
&nbsp;      uname='linux k83 2\.6\.32\-5\-amd64 \#1 smp mon oct 3 03&#8203;:59&#8203;:20 utc 2011 x86\_64 gnulinux '
&nbsp;      config\_args='\-Dprefix=/home/src/perl/repoperls/installed\-perls/perl/v5\.15\.4\-249\-ga03f0b9/a2da \-Dmyhostname=k83 \-Dinstallusrbinperl=n \-Uversiononly \-Dusedevel \-des \-Ui\_db \-Duseithreads \-Duselongdouble \-DDEBUGGING=\-g'
&nbsp;      hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;      useithreads=define\, usemultiplicity=define
&nbsp;      useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;      use64bitint=define\, use64bitall=define\, uselongdouble=define
&nbsp;      usemymalloc=n\, bincompat5005=undef
&nbsp;    Compiler&#8203;:
&nbsp;      cc='cc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
&nbsp;      optimize='\-O2 \-g'\,
&nbsp;      cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'
&nbsp;      ccversion=''\, gccversion='4\.4\.5'\, gccosandvers=''
&nbsp;      intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678
&nbsp;      d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;      ivtype='long'\, ivsize=8\, nvtype='long double'\, nvsize=16\, Off\_t='off\_t'\, lseeksize=8
&nbsp;      alignbytes=16\, prototype=define
&nbsp;    Linker and Libraries&#8203;:
&nbsp;      ld='cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;      libpth=/usr/local/lib /lib/\.\./lib /usr/lib/\.\./lib /lib /usr/lib /lib64 /usr/lib64
&nbsp;      libs=\-lnsl \-ldb \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
&nbsp;      perllibs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
&nbsp;      libc=/lib/libc\-2\.11\.2\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;      gnulibc\_version='2\.11\.2'
&nbsp;    Dynamic Linking&#8203;:
&nbsp;      dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
&nbsp;      cccdlflags='\-fPIC'\, lddlflags='\-shared \-O2 \-g \-L/usr/local/lib \-fstack\-protector'


&nbsp;  Characteristics of this binary \(from libperl\)&#8203;: 
&nbsp;    Compile\-time options&#8203;: HAS\_TIMES MULTIPLICITY PERLIO\_LAYERS
&nbsp;                          PERL\_DONT\_CREATE\_GVSV PERL\_IMPLICIT\_CONTEXT
&nbsp;                          PERL\_MALLOC\_WRAP PERL\_PRESERVE\_IVUV PERL\_USE\_DEVEL
&nbsp;                          USE\_64\_BIT\_ALL USE\_64\_BIT\_INT USE\_ITHREADS
&nbsp;                          USE\_LARGE\_FILES USE\_LOCALE USE\_LOCALE\_COLLATE
&nbsp;                          USE\_LOCALE\_CTYPE USE\_LOCALE\_NUMERIC USE\_LONG\_DOUBLE
&nbsp;                          USE\_PERLIO USE\_PERL\_ATOF USE\_REENTRANT\_API
&nbsp;    Built under linux
&nbsp;    Compiled at Nov 21 2011 09&#8203;:13&#8203;:37
&nbsp;    @&#8203;INC&#8203;:
&nbsp;      /home/src/perl/repoperls/installed\-perls/perl/v5\.15\.4\-249\-ga03f0b9/a2da/lib/site\_perl/5\.15\.4/x86\_64\-linux\-thread\-multi\-ld
&nbsp;      /home/src/perl/repoperls/installed\-perls/perl/v5\.15\.4\-249\-ga03f0b9/a2da/lib/site\_perl/5\.15\.4
&nbsp;      /home/src/perl/repoperls/installed\-perls/perl/v5\.15\.4\-249\-ga03f0b9/a2da/lib/5\.15\.4/x86\_64\-linux\-thread\-multi\-ld
&nbsp;      /home/src/perl/repoperls/installed\-perls/perl/v5\.15\.4\-249\-ga03f0b9/a2da/lib/5\.15\.4
&nbsp;      \.


\-\- 
andreas

**Comment by p5pRT** [comment]
### From @khwilliamson

I had failed to update installperl to install this new file\.
\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

I had failed to update installperl to install this new file\.
\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'new' to 'resolved'

**Comment by p5pRT** [comment]
### From @khwilliamson

On Mon Nov 21 16&#8203;:47&#8203;:38 2011\, khw wrote&#8203;:
> I had failed to update installperl to install this new file\.

And now fixed by 971c5244800c9017603979e4f5202b8e5e374b62

\-\- 
Karl Williamson
